pkg/apis/environment/basic.go

```diff
@@ -46,6 +46,24 @@ func (h Handler) Create(req CreateRequest) (CreateResponse, error) {
 
 		entity.Edges.Services = services
 
+		variableInputs := make(model.Variables, 0, len(req.Variables))
+
+		for _, rv := range req.Variables {
+			variable := rv.Model()
+			variable.ProjectID = entity.ProjectID
+			variable.EnvironmentID = entity.ID
+			variableInputs = append(variableInputs, variable)
+		}
+
+		variables, err := tx.Variables().CreateBulk().
+			Set(variableInputs...).
+			Save(req.Context)
+		if err != nil {
+			return err
+		}
+
+		entity.Edges.Variables = variables
```

Comment:
Can you use a function `EnvironmentVariablesEdgeSave` like `dao.EnvironmentConnectorsEdgeSave` and use it in SaveE?

Reply:
Moved variables save logic to dao. Moving service creation to dao will cause import cycle, so leave it for now.